frontend/src/actions/events.js

```diff
@@ -66,14 +77,50 @@ export const participantLeave = async (dispatch, eventId, userId, username) => {
   }
 };
 
-export const deleteEvent = async (dispatch, eventId) => {
+export const createEvent = (data) => async (dispatch) => {
   try {
-    const res = await genericApi.delete(`/api/events/${eventId}`);
+    genericApi.post("/api/events", data)
+      .then((res) => {
+        dispatch({
+          type: "CREATE_EVENT",
+          payload: res.data,
+        });
+      })
+      .then(() => {
+        dispatch({
+          type: "SHOW_SNACKBAR",
```

Comment:
I notice you dispatch(showSnackbar(..., ...) in some places and then dispatch({ type: SHOW_SNACKBAR...}) in some other places. Is there a reason you dont just use showSnackbar action? I think its better to use showSnackbar so if we change the type SHOW_SNACKBAR to a different name, you change once in action and once in reducer as oppose to every single snackbar dispatch.

Reply:
copy pasted, can change